### ISSUE TITLE
Problem with refreshing menu items for restricted account permissions

### ISSUE THREAD

**Issue by momus300** [marmelab/react-admin#7586]
Hi, I found a problem with refreshing resource menu between logging into account with restricted permission (admin to user).

**What you were expecting:**
Refreshing resource menu between logging into different accounts.

**What happened instead:**
Account does not have a restricted menu.

**Steps to reproduce:**
- Set restricted resource for e.g. `admin` role:
```
      {(permissions) => (
        <>
          {permissions === "admin" ? (
            <Resource name="users" {...users} />
          ) : null}
```
- Login to admin account
- logout
- login to user account
- resource menu still contains `users` item (it will change after browser refresh):
![image](https://user-images.githubusercontent.com/5521530/164893722-c3ae36e1-b373-4fbc-8906-c08ad9aec874.png)

**Related code:**

* CodeSandbox:
  - https://codesandbox.io/s/stoic-cache-on72hn?file=/src/index.tsx

**Environment**

* React-admin version: 4.0.1
* Last version that did not exhibit the issue (if applicable): 3.19
* React version: 18.0
* Browser: Chrome

**Comment by slax57** [comment]
Hi @momus300 
Indeed you are right, this problem was identified while working on #7527 and will hopefully be solved with #7539
I'm keeping your issue open since #7527 was closed, and will link it to #7539

**Comment by slax57** [comment]
Fixed via #7539
